src/Template/Deserialization.php

```diff
@@ -62,7 +62,16 @@ function {{functionName}}(array {{jsonPath}}): {{className}}
 EOT;
 
     private const TMPL_ASSIGN_DATETIME_FROM_FORMAT = <<<'EOT'
-{{modelPath}} = \DateTime::createFromFormat('{{format}}', {{jsonPath}});
+foreach([{{formats|join(', ')}}] as {{formatVariable}}) {
+    if (({{formatVariable}} = \DateTime::createFromFormat({{formatVariable}}, {{jsonPath}}, {{timezone}}))) {
+        {{modelPath}} = {{formatVariable}};
```

Comment:
same question here, can't we call the temporary variable just `date` or something like that?